src/partition/repart.c

```diff
@@ -6113,6 +6289,171 @@ static int fd_apparent_size(int fd, uint64_t *ret) {
         return 0;
 }
 
+static bool need_fstab_one(const Partition *p) {
+        assert(p);
+
+        if (p->dropped)
+                return false;
+
+        if (!p->format)
+                return false;
+
+        if (p->n_mountpoints == 0)
+                return false;
+
+        return true;
+}
+
+static bool need_fstab(const Context *context) {
+        assert(context);
+
+        if (!arg_generate_fstab)
+                return false;
+
+        LIST_FOREACH(partitions, p, context->partitions)
+                if (need_fstab_one(p))
+                        return true;
+
+        return false;
+}
+
+static int context_fstab(Context *context) {
+        _cleanup_(unlink_and_freep) char *t = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r;
+
+        assert(context);
+
+        if (!need_fstab(context))
+                return 0;
```

Comment:
it feels a bit weird to not generate any fstab here, event though the user explicitly said so via the command line.

maybe just add a log_notice() message here, i.e. "No suitable partitions carry MountPoint=", not generating %s."

I think that would be less confusing for users. It's not an error after all, but still deserves notice.